x/eibc/keeper/msg_server.go

```diff
@@ -72,6 +72,127 @@ func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrde
 	return &types.MsgFulfillOrderResponse{}, nil
 }
 
+func (m msgServer) FulfillOrderAuthorized(goCtx context.Context, msg *types.MsgFulfillOrderAuthorized) (*types.MsgFulfillOrderAuthorizedResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := ctx.Logger()
+
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	demandOrder, err := m.GetOutstandingOrder(ctx, msg.OrderId)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := m.validateOrder(demandOrder, msg, ctx); err != nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrUnauthorized, err.Error())
+	}
+
+	granterAccount := m.ak.GetAccount(ctx, msg.GetLPBech32Address())
+	if granterAccount == nil {
+		return nil, types.ErrGranterAddressDoesNotExist
+	}
+
+	// Send the funds from the granterAccount to the eibc packet original recipient
+	err = m.bk.SendCoins(ctx, granterAccount.GetAddress(), demandOrder.GetRecipientBech32Address(), demandOrder.Price)
+	if err != nil {
+		logger.Error("Failed to send price to recipient", "error", err)
+		return nil, err
+	}
+
+	// TODO: will this work for Policy address?
+	fulfillerAccount := m.ak.GetAccount(ctx, msg.GetOperatorBech32Address())
+	if fulfillerAccount == nil {
+		return nil, types.ErrFulfillerAddressDoesNotExist
+	}
+
+	// by default, the operator account receives the operator share
+	feePartReceiver := fulfillerAccount
+	// if the operator fee address is provided, the operator fee share is sent to that address
+	if msg.OperatorFeeAddress != "" {
+		operatorFeeAccount := m.ak.GetAccount(ctx, msg.GetOperatorFeeBech32Address())
+		if operatorFeeAccount == nil {
+			return nil, types.ErrOperatorAddressDoesNotExist
+		}
+		feePartReceiver = operatorFeeAccount
+	}
+
+	fee, _ := sdk.NewDecFromStr(demandOrder.Fee.String())
```

Comment:
maybe I'm missing something but not sure where do we validate that the lp fee is at least as min_fee (i.e order_fee(nominal) - operator_fee(nominal) > min_fee% * order_price)

Reply:
https://github.com/dymensionxyz/dymension/blob/6194954c0356586af5187dd184e9d6050a927608/x/eibc/types/fulfill_order_authorization.go#L91